### PR TITLE
refactor: move i18n service to npm

### LINKED ISSUE #6173: move i18n plugin to its own npm package, and install npm version Reaction
New repo: https://github.com/reactioncommerce/api-plugin-i18n  

Convert the `i18n` plugin. This includes moving all related files into a new repository, releasing that repo as an `npm` package, and then updating this repo, `reaction`, to use the `npm` version of the plugin.

Part of [this epic](https://github.com/reactioncommerce/reaction/issues/6107) to move all plugins into their own repositories and release as `npm` packages.

### PR DESCRIPTION
**Pull request by mpaktiti**
Resolves https://github.com/reactioncommerce/reaction/issues/6173

This PR removes the internal `i18n` service and installs the new [api-plugin-i18n npm package](https://www.npmjs.com/package/@reactioncommerce/api-plugin-i18n) as its replacement.

The code in this npm package is the same, and has already been tested, so the expected outcome of this PR is seeing nothing different than how the app currently runs.

Signed-off-by: Maria Paktiti <maria.paktiti@gmail.com>

### CODE REVIEW

File: src/registerPlugins.js [hunk 1]
```diff
@@ -10,14 +10,14 @@ import registerPluginSimpleAuthorization from "@reactioncommerce/plugin-simple-a
 import registerPluginSitemapGenerator from "@reactioncommerce/plugin-sitemap-generator";
 import registerPluginSystemInformation from "@reactioncommerce/plugin-system-information";
 import registerPluginTranslations from "@reactioncommerce/plugin-translations";
+import registerPluginI18n from "@reactioncommerce/api-plugin-i18n";
```
Comment (kieckhafer) [inline src/registerPlugins.js]: Please alphabetize this for organizational purposes.

File: jest.config.cjs [hunk 2]
```diff
@@ -16,7 +16,8 @@ const externalNodeModules = [
   "@reactioncommerce/plugin-simple-authorization",
   "@reactioncommerce/plugin-sitemap-generator",
   "@reactioncommerce/plugin-system-information",
-  "@reactioncommerce/plugin-translations"
+  "@reactioncommerce/plugin-translations",
+  "@reactioncommerce/api-plugin-i18n"
```
Comment (kieckhafer) [inline jest.config.cjs]: Please alphabetize this for organizational purposes.

File: package.json [hunk 3]
```diff
@@ -22,6 +22,7 @@
   "dependencies": {
     "@apollo/federation": "^0.12.0",
     "@reactioncommerce/api-plugin-accounts": "~1.2.0",
+    "@reactioncommerce/api-plugin-i18n": "^1.0.0",
```
Comment (kieckhafer) [inline package.json]: Lets use the `~` here for the versioning like we do with the other api imports.

**Review by kieckhafer — CHANGES_REQUESTED** [review]
Looks good overall, a few small things

File: jest.config.cjs [hunk 4]
```diff
@@ -9,6 +9,7 @@ const externalNodeModules = [
   "@reactioncommerce/api-plugin-accounts",
   "@reactioncommerce/api-plugin-address-validation",
   "@reactioncommerce/api-plugin-address-validation-test",
+  "@reactioncommerce/api-plugin-i18n",
```
Comment (mpaktiti) [inline jest.config.cjs:10]: @kieckhafer should I also move lines 7+8 under the `api-plugin-i18n` or they are left there for a reason?

Comment (kieckhafer) [reply jest.config.cjs:10]: No reason, I updated them.

**Review by kieckhafer — APPROVED** [review]
👍 changes approved

**Pull request merged**